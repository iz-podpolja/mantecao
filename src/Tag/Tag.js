import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors, theme } from '../variables'

import CloseIcon from '../icons/CloseIcon'

const CloseSpan = styled.span`
  cursor: pointer;
  display: inline-block;
  margin-left: 0.35em;

  &:hover {
    color: rgba(188, 33, 63, 1);
  }
`

const TagDiv = styled.div`
  background-color: ${props =>
    props.hovered ? 'rgba(188, 33, 63, 0.2)' : 'rgba(248,193,80, 0.2)'};
  border: ${props =>
    props.hovered
      ? '1px solid rgba(248,193,80, 0.2)'
      : '1px solid rgba(188, 33, 63, 0.3)'};
  border-left: 4px solid;
  border-left-color: ${props =>
    props.hovered ? 'rgba(188, 33, 63, 1)' : colors.goldenTainoi};
  border-radius: 2px;
  color: ${props => (props.hovered ? colors.emperor : colors.dustyGray)};
  display: inline-block;
  font-size: 0.8rem;
  padding-right: 0.7em;
  font-family: ${theme.fontFamily};
  line-height: ${theme.lineHeight};
`

const TagSpan = styled.span`
  display: inline-block;
  padding: 0.5em 0.1em 0.5em 0.8em;
`

const AdjustedCloseIcon = (
  <CloseIcon
    style={{
      height: '1.2em',
      width: '1.2em',
      strokeWidth: '2.5px',
      marginBottom: '0.2em'
    }}
  />
)

class Tag extends Component {
  state = {
    hovered: false
  }

  render() {
    const { name, onClickAction } = this.props

    return (
      <TagDiv hovered={this.state.hovered}>
        <TagSpan>{name}</TagSpan>
        {onClickAction &&
          <CloseSpan
            onClick={() => onClickAction(name)}
            onMouseEnter={() => {
              this.setState({ hovered: true })
            }}
            onMouseLeave={() => {
              this.setState({ hovered: false })
            }}
          >
            {AdjustedCloseIcon}
          </CloseSpan>}
      </TagDiv>
    )
  }
}

Tag.propTypes = {
  name: PropTypes.string.isRequired,
  onClickAction: PropTypes.func
}

export default Tag
