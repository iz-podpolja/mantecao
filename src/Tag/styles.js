import { css } from 'glamor'
import { colors } from '@resmio/rollico/dist/'

export const styles = {
  tag: css({
    backgroundColor: 'rgba(248,193,80,.2)',
    border: '1px solid rgba(248,193,80,.3)',
    borderLeft: `4px solid ${colors.goldenTainoi}`,
    borderRadius: '2px',
    color: colors.dustyGray,
    display: 'inline-block',
    fontSize: '12px',
    padding: '.5rem .8rem'
  }),

  close: css({
    cursor: 'pointer',
    display: 'inline-block',
    marginLeft: '0.75rem',
    ':hover': {
      color: 'rgba(188, 33, 63, 1)'
    }
  }),

  hover: css({
    background: 'rgba(188, 33, 63, 0.2)',
    color: colors.emperor,
    border: '1px solid rgba(188, 33, 63, 0.3)',
    borderLeft: '4px solid rgba(188, 33, 63, 1)'
  })
}
