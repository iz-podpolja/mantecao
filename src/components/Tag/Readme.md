With close:

    <Tag
      name={'tag 1'}
      onClickAction={() => console.log('tag clicked')}
    />

without:

    <Tag
      name={'donotremove'}
    />
