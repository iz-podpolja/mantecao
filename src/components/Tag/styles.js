import { css } from 'glamor'
import {colors} from '@resmio/rollico/dist/'

export const styles = {
  tag: css({
    backgroundColor: 'rgba(248,193,80,.2)',
    border: '1px solid rgba(248,193,80,.3)',
    borderLeft: `4px solid ${colors.goldenTainoi}`,
    borderRadius: '2px',
    color: colors.dustyGray,
    display: 'inline-block',
    fontSize: '12px',
    padding: '.5rem .8rem'
  }),

  close: css({
    ':hover': {
      color: 'rgba(188, 33, 63, 1)',
      cursor: 'pointer'
    }
  }),

 hover: css({
    background: 'rgba(188, 33, 63, 0.2)',
    color: colors.emperor,
    border: '1px solid rgba(188, 33, 63, 0.3)',
    borderLeft: `4px solid rgba(188, 33, 63, 1)`,
  })
}
