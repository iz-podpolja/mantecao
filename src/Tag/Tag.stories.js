import React from 'react'
import { storiesOf } from '@storybook/react'
import Tag from './Tag'

storiesOf('Tag', module)
  .add('with close', () =>
    <div>
      <Tag name="tag 1" onClickAction={() => console.log('tag clicked')} />
      <Tag name="donotremove" />
    </div>
  )
  .add('without close', () => <Tag name="donotremove" />)
