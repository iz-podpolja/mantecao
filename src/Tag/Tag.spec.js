import React from 'react'
import renderer from 'react-test-renderer'
import Tag from './Tag'

const ON_CLICK_ACTION = jest.fn()

describe('Tag Component', () => {
  it('renders with no close button if not onCLickAction prop is passed', () => {
    const tree = renderer.create(<Tag name="tag 1" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with a close button if an onCLickAction prop is passed', () => {
    const tree = renderer
      .create(<Tag name="tag 1" onClickAction={ON_CLICK_ACTION} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  // TODO
  it('fires the onClickAction function when the close button is pressed')
  it('changes the class of the component when the close button is hovered')
})
