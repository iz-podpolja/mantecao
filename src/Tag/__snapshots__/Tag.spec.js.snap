// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component renders with a close button if an onCLickAction prop is passed 1`] = `
<div
  className="sc-bwzfXH cgmXoR"
>
  <span
    className="sc-htpNat jbZKpL"
  >
    tag 1
  </span>
  <span
    className="sc-bdVaJa cJwrfy"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      className={undefined}
      style={
        Object {
          "display": "inline-block",
          "fill": "none",
          "height": "1.2em",
          "marginBottom": "0.2em",
          "pointerEvents": "none",
          "stroke": "currentColor",
          "strokeWidth": "2.5px",
          "verticalAlign": "middle",
          "width": "1.2em",
        }
      }
      viewBox="0 0 32 32"
    >
      <g
        transform={undefined}
      >
        <g
          strokeLinecap="square"
        >
          <path
            d="M5.5 6.5L25.5 26.5M25 7L5 27"
          />
        </g>
      </g>
    </svg>
  </span>
</div>
`;

exports[`Tag Component renders with no close button if not onCLickAction prop is passed 1`] = `
<div
  className="sc-bwzfXH cgmXoR"
>
  <span
    className="sc-htpNat jbZKpL"
  >
    tag 1
  </span>
</div>
`;
